src/bond/assertions.clj

```diff
@@ -14,6 +14,23 @@
 (defn called-with-args?
   "An assertion to check if `f` was called with `vargs`.
 
-  `vargs` should be a vector of args [args-first-call args-second-call ...] to allow for the checking of multiple calls of `f`."
+  `vargs` should be a vector of args [args-first-call args-second-call ...] to allow for the checking of multiple calls of `f`.
+   Note that this method asserts about every call to `f`."
   [f vargs]
   (= vargs (->> f bond/calls (mapv :args))))
+
+(defn called-once-with-args?
+  "An assertion to check if `f` was called with `args` strictly once.
+   
+   `args` should be a vector/coll of args [arg1 arg2 arg3] to compare directly to the value of `:args` from `bond/calls`"
+  [f args]
+  (and (called-times? f 1)
```

Comment:
i think the call to `called-times?` is not necessary? since having args in a vector of length 1 implies it is called once

Reply:
Ooh, that's a good point.

At that point, maybe `called-once-with-args` isn't needed since it just becomes a wrapper for `called-with-args`?

Reply:
Unless the extra specificity in the name is useful

Reply:
i don't mind functions that simply wrap other functions when they significantly improve the readability/understanding of the code being wrapped.

```clj
(called-with-args? foo [[1 2 3]])
(called-once-with-args? foo [1 2 3])
```

i think the latter is much more clearly asserting it's called once with `[1 2 3]` as args.

as a user of the library, how often do you find yourself wanting to check that a thing is called exactly once? if it's pretty often, i think a convenience function makes sense